pkgs/test/texlive/default.nix

```diff
@@ -119,4 +119,37 @@
     texdoc --debug --list texdoc | tee "$out"
     grep texdoc.pdf "$out"
   '';
+
+  # test that language files are generated as expected
+  hyphen-base = runCommandNoCC "texlive-test-hyphen-base" {
+    hyphenBase = lib.head texlive.hyphen-base.pkgs;
+    schemeFull = texlive.combined.scheme-full;
+    schemeInfraOnly = texlive.combined.scheme-infraonly;
+  } ''
+    mkdir -p "$out"/{scheme-infraonly,scheme-full}
+
+    # create language files with no hyphenation patterns
+    cat "$hyphenBase"/tex/generic/config/language.us >language.dat
+    cat "$hyphenBase"/tex/generic/config/language.us.def >language.def
+    cat "$hyphenBase"/tex/generic/config/language.us.lua >language.dat.lua
+
+    cat >>language.dat.lua <<EOF
+    }
+    EOF
+
+    cat >>language.def <<EOF
+    %%% No changes may be made beyond this point.
+
+    \uselanguage {USenglish}             %%% This MUST be the last line of the file.
+    EOF
+
+    for fname in language.{dat,def,dat.lua} ; do
+      diff --ignore-matching-lines='^\(%\|--\) Generated by ' -u \
+        {"$hyphenBase","$schemeFull"/share/texmf}/tex/generic/config/"$fname" \
+        | tee "$out/scheme-full/$fname.patch"
+      diff --ignore-matching-lines='^\(%\|--\) Generated by ' -u \
+        {,"$schemeInfraOnly"/share/texmf/tex/generic/config/}"$fname" \
+        | tee "$out/scheme-infraonly/$fname.patch"
```

Comment:
Is there a practical observable effect of correctly generating the language files? If there is, I would test for that instead.

Reply:
Right. This is just the test I was doing by hand. There are indeed side effects, e.g. https://tug.org/pipermail/tex-live/2021-July/047229.html. On the other hand, I don't know how to test every language (62 packages plus synonyms) and catch if sed mangles some attributes from side effects only. I'll think about it.

Reply:
Uhm, the best I can do is run `pdflatex` and check that all languages are available. That should confirm that `language.{dat,def}` are mostly correct, but it will not notice if the footer is missing (the comments do not say what it does!). For `language.dat.lua`, it is proving too tricky, I am giving up.

Shall we keep this `diff` test? I think it's a good one, anyway.

Reply:
If there isn't an obvious test, it's fine to stick with this.

Reply:
Actually, I have some decent tests now for all three files. For example, I can test whether US English is the default language in `etex` and `luatex` (currently it's not!). But it's five or six tests and I need to polish them, and I don't want to delay this PR right now.